python/indexing/regex_mlir_bindings.py

```diff
@@ -0,0 +1,61 @@
+import re
+import sys
+from os.path import exists
+
+symbol_guard = """\
+#ifdef MY_PYBIND11_EXPORT_SYMBOLS
+#  define MY_PYBIND11_EXPORT __attribute__((visibility("default")))
+#else
+#  define MY_PYBIND11_EXPORT
+#endif
+"""
+
+module_guard = """\
+#ifdef MY_PYBIND11_EXPORT_MODULE
+#  define MY_PYBIND11_LOCAL_MODULE
+#else
+#  define MY_PYBIND11_LOCAL_MODULE , py::module_local()
+#endif
+"""
+
+def ir_module(fp):
+  with open(fp) as f:
+    f = f.read()
+  if not exists(f"{fp}.bkup"):
+    with open(f"{fp}.bkup", "w") as bkup:
+      bkup.write(f)
+    with open(fp, "w") as ff:
+      ff.write(symbol_guard)
+      res = re.sub(
+          r"^class ((?!MY_PYBIND11_EXPORT).*) {$",
+          r"class MY_PYBIND11_EXPORT \1 {",
+          f,
+          flags=re.MULTILINE,
+      )
+      res = res.replace(
+          "class DefaultingPyMlirContext",
+          "class MY_PYBIND11_EXPORT DefaultingPyMlirContext",
+      )
```

Comment:
See [this comment](https://github.com/iree-org/iree-llvm-sandbox/pull/687/files#r1166204774).